src/Workspaces/Core/Portable/Editing/SyntaxEditor.cs

```diff
@@ -13,6 +13,8 @@ public class SyntaxEditor
     {
         private readonly SyntaxGenerator _generator;
         private readonly List<Change> _changes;
+        private bool _allowEditsOnLazilyCreatedTrackedNewNodes;
+        private HashSet<SyntaxNode> _lazyTrackedNewNodesOpt;
```

Comment:
question.  why not always do this?

Reply:
Yes, perf was my primary concern, but I am fine with this being the default if @dotnet/roslyn-ide agree and we can switch back to it being opt-in later if needed.